docs/zkapps/tutorials/10-account-updates.mdx

```diff
@@ -0,0 +1,381 @@
+      setZkappUri: Permissions.proof(),
+      setTokenSymbol: Permissions.proof(),
+      incrementNonce: Permissions.proof(),
+      setVotingFor: Permissions.proof(),
+      setTiming: Permissions.proof(),
+    });
+  }
+
+  @method init() {
+    this.account.provedState.assertEquals(this.account.provedState.get());
+    this.account.provedState.get().assertFalse();
+
+    super.init();
+    this.num.set(Field(1));
+    this.calls.set(Field(0));
+  }
+
+  ...
+```
+
+This configures the zkApp as described, and initializes the zkApp with the values we'd like. 
+
+:::caution
+Note we assert that `provedState` is `False` in `init`. This ensures that `init` cannot be called again after the zkApp is setup. Without this, our zkApp could be reset by anyone calling the `init` transaction on our zkApp, and is recommended for most zkApps.
+:::
+
+Next, we'll add two functions:
+
+```typescript
+  ...
+
+  @method add(incrementBy: Field) {
+    this.account.provedState.assertEquals(this.account.provedState.get());
+    this.account.provedState.get().assertTrue();
+
+    const num = this.num.get();
+    this.num.assertEquals(num);
+    this.num.set(num.add(incrementBy));
+
+    this.incrementCalls();
+  }
+
+  @method incrementCalls() {
+    this.account.provedState.assertEquals(this.account.provedState.get());
+    this.account.provedState.get().assertTrue();
+
+    const calls = this.calls.get();
+    this.calls.assertEquals(calls);
+    this.calls.set(calls.add(Field(1)));
+  }
+
+  ...
+```
+
+:::caution
+In each of these, we are also asserting `provedState` is `true`. This ensures the zkApp was initialized as expected, as `provedState` becomes true after `init` is called, and is recommended for most zkApps.
+:::
+
+Note we are also having the `add` method call the `incrementCalls` method, to see how this is reflected in an `add` transaction's AccountUpdate structure.
+
+Lastly, we'll add one more function, `callSecondary`, which calls a different zkApp:
+
+```typescript
+  ...
+
+  @method callSecondary(secondaryAddr: PublicKey) {
+    this.account.provedState.assertEquals(this.account.provedState.get());
+    this.account.provedState.get().assertTrue();
+
+    const secondaryContract = new SecondaryZkApp(secondaryAddr);
+
+    const num = this.num.get();
+    this.num.assertEquals(num);
+
+    secondaryContract.add(num);
+
+    // NOTE this gets the state at the start of the transaction
+    this.num.set(secondaryContract.num.get());
+
+    this.incrementCalls();
+  }
+}
+```
+
+This takes the address of our other zkApp, `SecondaryZkApp`, and calls a method on it. Note that the impact of calling that method will occur after this set of AccountUpdates - so when we call `secondaryContract.num.get()`, that will get the value before this transaction was applied.
+
+Last, to look briefly at [SecondaryZkApp.ts](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/10-account-updates/src/SecondaryZkApp.ts), we have contents:
```

Comment:
```suggestion
This takes the address of our other zkApp, `SecondaryZkApp`, and calls a method on it. Note that the impact of calling that method occurs after this set of AccountUpdates - so when you call `secondaryContract.num.get()` that gets the value before this transaction is applied.

Last, to look briefly at [SecondaryZkApp.ts](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/10-account-updates/src/SecondaryZkApp.ts), you have contents:
```

Reply:
@es92 present tense https://developers.google.com/style/tense 

please verify I did not inadvertently change the technical accuracy :)